### DISCUSSION TITLE
@babel/preset-typescript for react >= 17

### DISCUSSION THREAD

**Discussion by ym-project** [babel/babel#14728] [Q&A]
Hi!

Before react < 17 I had the next babel config:
```js
presets: [
    ['@babel/preset-env', {...}],
    ['@babel/preset-react', {pragma: 'React.createElement', pragmaFrag: 'React.Fragment'}],
    ['@babel/preset-typescript', {jsxPragma: 'React', jsxPragmaFrag: 'React.Fragment'}],
],
```
And it worked perfectly.

Now I try to migrate to react >= 17 and I can find nothing about `@babel/preset-typescript` for new react versions. I have new babel config file:
```
presets: [
    ['@babel/preset-env', {...}],
    ['@babel/preset-react', {runtime: "automatic"}],
    ['@babel/preset-typescript', {??????}]
]
```

What should I pass to `@babel/preset-typescript` preset options?

**Comment by nicolo-ribaudo** [comment]
Nothing, it should just work. What errors do you see?

**Reply by ym-project** [reply]
There is no errors because I use `jsx: "preserve"` option inside `tsconfig.json`. But I'm interesting in how to manually rewrite preset pragma?
